[ERROR] Failed to fix nested if: Expected outer if to have indented body and no else

ruff version: `0.0.230`

```python
a = True
b = True

if a:
    if b:
        print('do something')  # noqa: T201
else:
    print('at else')  # noqa: T201
```

Running ruff here will result with:

```
test.py:4:1: SIM102 Use a single `if` statement instead of nested `if` statements
```

If we try to autofix we get `[2023-01-22][23:09:59][ruff::rules::flake8_simplify::rules::ast_if][ERROR] Failed to fix nested if: Expected outer if to have indented body and no else`

--- a/resources/test/fixtures/flake8_simplify/SIM102.py
+++ b/resources/test/fixtures/flake8_simplify/SIM102.py
@@ -92,3 +92,40 @@ if node.module:
         "multiprocessing."
     ):
         print("Bad module!")
+
+
+# OK
+if a:
+    if b:
+        print("foo")
+else:
+    print("bar")
+
+# OK
+if a:
+    if b:
+        if c:
+            print("foo")
+        else:
+            print("bar")
+else:
+    print("bar")
+
+# OK
+if a:
+    # SIM 102
+    if b:
+        if c:
+            print("foo")
+else:
+    print("bar")
+
+
+# OK
+if a:
+    if b:
+        if c:
+            print("foo")
+        print("baz")
+else:
+    print("bar")

--- a/src/checkers/ast.rs
+++ b/src/checkers/ast.rs
@@ -1386,6 +1386,7 @@ where
                         stmt,
                         test,
                         body,
+                        orelse,
                         self.current_stmt_parent().map(Into::into),
                     );
                 }

--- a/src/rules/flake8_simplify/rules/ast_if.rs
+++ b/src/rules/flake8_simplify/rules/ast_if.rs
@@ -50,7 +50,7 @@ fn is_main_check(expr: &Expr) -> bool {
 /// ```
 fn find_last_nested_if(body: &[Stmt]) -> Option<(&Expr, &Stmt)> {
     let [Stmt { node: StmtKind::If { test, body: inner_body, orelse }, ..}] = body else { return None };
-    if !(orelse.is_empty() && body.len() == 1) {
+    if !orelse.is_empty() {
         return None;
     }
     find_last_nested_if(inner_body).or_else(|| {
@@ -67,12 +67,10 @@ pub fn nested_if_statements(
     stmt: &Stmt,
     test: &Expr,
     body: &[Stmt],
+    orelse: &[Stmt],
     parent: Option<&Stmt>,
 ) {
-    if is_main_check(test) {
-        return;
-    }
-
+    // If the parent could contain a nested if-statement, abort.
     if let Some(parent) = parent {
         if let StmtKind::If { body, orelse, .. } = &parent.node {
             if orelse.is_empty() && body.len() == 1 {
@@ -81,6 +79,16 @@ pub fn nested_if_statements(
         }
     }
 
+    // If this if-statement has an else clause, or more than one child, abort.
+    if !(orelse.is_empty() && body.len() == 1) {
+        return;
+    }
+
+    if is_main_check(test) {
+        return;
+    }
+
+    // Find the deepest nested if-statement, to inform the range.
     let Some((test, first_stmt)) = find_last_nested_if(body) else {
         return;
     };

